backend/app/routers/experiment_runs.py

```diff
@@ -78,7 +82,69 @@ async def get_experiment_run(
     user: dict = Depends(get_current_user(required=False)),
 ) -> Any:
     experiment_run = await get_experiment_run_if_accessible_or_raise(id, user)
-    return experiment_run.map_to_response(return_detailed_response=True)
+    experiment = await Experiment.get(experiment_run.experiment_id)
+
+    is_mine = user is not None and experiment.created_by == user["email"]
+    return experiment_run.map_to_response(
+        is_mine=is_mine, return_detailed_response=True
+    )
+
+
+@router.get("/experiment-runs/{id}/stop", response_model=None)
+async def stop_experiment_run(
+    id: PydanticObjectId,
+    user: dict = Depends(get_current_user(required=True)),
+    workflow_engine: WorkflowEngineBase = Depends(ReanaService.get_service),
+) -> Any:
+    # TODO this case needs to be properly addressed
+    # TODO if this endpoint is executed even before the workflow has even started
+    # (the code execution is in _general_workflow_preparation function for example),
+    # it will not effectively not stop the experiment run execution pipeline and
+    # the workflow will be created later on
+
+    experiment_run = await get_experiment_run_if_accessible_or_raise(
+        id, user, write_access=True
+    )
+
+    experiment = await Experiment.get(experiment_run.experiment_id)
+    if experiment.is_archived:
```

Comment:
Can you explain why this need to be here? I suppose you cannot run an archived experiment and cannot archive experiment that is running.

Reply:
Yeah, I suppose once the experiment is archived, you cannot perform any changes to its attributes and its runs. Frankly, I haven't considered the case when there are some experiment runs still ongoing once you archive an experiment. **I wonder whether we should be able to archive an experiment only in the case there are no experiment runs running.** 

For now though, you can archive an experiment with currently running runs and these runs will not be terminated